src/state.rs

```diff
@@ -107,6 +107,13 @@ impl BarState {
         self.update_estimate_and_draw(now);
     }
 
+    pub(crate) fn dec_length(&mut self, now: Instant, delta: u64) {
```

Comment:
Nit: for consistency, let's order `dec_length()` after `inc_length()` to match the ordering on all the related methods.